arch/arm/core/aarch64/reset.S

```diff
@@ -44,60 +49,89 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__reset)
 GTEXT(__start)
 SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 
-#ifdef CONFIG_SWITCH_TO_EL1
-	switch_el x1, 3f, 2f, 1f
-3:
-	/* Disable MMU and async exceptions routing to EL1 */
-	msr	sctlr_el1, xzr
-
-	/* Disable EA/IRQ/FIQ routing to EL3 and set EL1 to AArch64 */
-	mov	x0, xzr
-	orr	x0, x0, #(SCR_EL3_RW)
-	msr	scr_el3, x0
-
-	/* On eret return to EL1 with DAIF masked */
-	mov	x0, xzr
-	orr	x0, x0, #(DAIF_MASK)
-	orr	x0, x0, #(SPSR_EL3_TO_EL1)
-	orr	x0, x0, #(SPSR_EL3_h)
-	msr	spsr_el3, x0
-
-	adr	x0, 1f
-	msr	elr_el3, x0
-	eret
-2:
-	/* Boot from EL2 not supported */
-	bl	.
-1:
-#endif
 	/* Setup vector table */
-	adr	x0, _vector_table
+	adr	x9, _vector_table
 
 	switch_el x1, 3f, 2f, 1f
 3:
 	/* Initialize VBAR */
-	msr	vbar_el3, x0
+	msr	vbar_el3, x9
+	isb
+
+	/* Initialize sctlr_el3 to reset value */
+	mov_imm	x2, SCTLR_EL3_RES1
+	mrs     x1, sctlr_el3
+	orr	x1, x1, x2
+	msr	sctlr_el3, x1
+	isb
 
 	/* SError, IRQ and FIQ routing enablement in EL3 */
-	mrs	x0, scr_el3
-	orr	x0, x0, #(SCR_EL3_IRQ | SCR_EL3_FIQ | SCR_EL3_EA)
-	msr	scr_el3, x0
+	mrs	x1, scr_el3
```

Comment:
Why are you avoiding to use the `x0` register?

Reply:
I kept on changing that couple of times and just thought will do as per review. Intention was to have x0 reserved for special purpose eg: provision to get argument passed from previous stage (if any). But If that happens that can be handled anyway.
  So now its more of choice. can fall back to x0.

Reply:
Yeah, that's a good thought but I would probably switch back to `x0` for this PR just to keep the diff size as minimal as possible.

Reply:
ACK .  I will push the new version with uniform usage of registers starting with x0 , x0 ...